src/logging/WriterBackend.h

```diff
@@ -38,9 +38,28 @@ class WriterBackend : public threading::MsgThread {
      * frontend, it's running in a different thread.
      *
      * @param name A descriptive name for writer's type (e.g., \c Ascii).
+     */
+    [[deprecated(
+        "Remove in v9.1: Use WriterBackend(NoThreadingHeartbeats{}, frontend). If you have a use for heartbeats in "
```

Comment:
Does that seem a reasonable deprecation path?

I found that [zeek-kafka](https://github.com/SeisoLLC/zeek-kafka/blob/0a0c44267b17ad274fd7a2af780739dbaa0b089f/src/KafkaWriter.cc#L304-L309) does call `producer->poll()`, but I'd think that should better from a thread started by that thread using real-time, rather than network-time.

Reply:
> The only problem I see is that it doesn't give a nice deprecation warning about it.

That was the main motivation of the constructor tagging: Defaulting to `false` would change behavior possibly without a plugin author or user realizing. I'm not sure how much `zeek-kafka` really relies on the `DoHeartBeat()` call, but was too timid to simply stop invoking it by default.

I could seek keeping the default as `true` and adapting Zeek's internal writers to pass `false` explicitly. That didn't force moving to a better place to me enough at the time (which the deprecation warning does), but I'm certainly okay with that if the constructor tagging seems like overdoing it.

Reply:
@timwoj - I have reworked this to just add a new constructor parameter that's defaulted to true though to not break existing writers. Mind taking another look?